pilot/proxy/envoy/discovery.go

```diff
@@ -427,6 +435,25 @@ func (ds *DiscoveryService) parseDiscoveryRequest(request *restful.Request) (pro
 	return role, nil
 }
 
+// AvailabilityZone responds to requests for an AZ for the given cluster node
+func (ds *DiscoveryService) AvailabilityZone(request *restful.Request, response *restful.Response) {
+	role, err := ds.parseDiscoveryRequest(request)
+	if err != nil {
+		errorResponse(response, http.StatusNotFound, "AvailabilityZone "+err.Error())
+		return
+	}
+	instances, err := ds.HostInstances(map[string]bool{role.IPAddress: true})
+	if err != nil {
+		errorResponse(response, http.StatusNotFound, "AvailabilityZone "+err.Error())
+		return
+	}
+	if len(instances) <= 0 {
+		errorResponse(response, http.StatusNotFound, "AvailabilityZone couldn't find a AZ for the given cluster node")
+	}
+	// All instances are going to have the same addr so will all be in the same AZ
+	writeResponse(response, []byte(instances[0].AvailabilityZone))
```

Comment:
Is it possible for us to hit cases where services on the same host report different AZs? In other words, is it possible for the notion of an AZ to vary per service (or, what if two different service registries have different notions of AZs and both have entries for services on the same host)?